### PR TITLE
[ISSUE #8090] Optimize isSetEqual for DefaultLitePullConsumerImpl

### LINKED ISSUE #8090: [Enhancement] 方法DefaultLitePullConsumerImpl#isSetEqual可以优化
### Before Creating the Enhancement Request

- [X] I have confirmed that this should be classified as an enhancement rather than a bug/feature.

### Summary

1. 方法isSetEqual在比较两个set时，两个set都为empty时，返回的是不相等，从逻辑上来说，应该是true吧，毕竟两个set都为null判定的都是相等。从代码上说，从broker获取topic的messagequeue不可能是空的，否则会抛MQClientException异常。因此我认为应该移除掉set1.size()==0这个判断。
image
image
image
2. 当已经找到了一个set1中存在，但是set2中不存在的元素时，就可以判定两个set不相等了，直接退出循环即可，没必要再继续比较了
image

### Motivation

优化isSetEqual以提高方法性能

### Describe the Solution You'd Like

1. 移除错误且不必要的分支判断set1.size() == 0
2. 检测到不相等元素时，提前退出循环并返回结果

### Describe Alternatives You've Considered

无

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Willhow-Gao**
Fixes #8090 
1. 优化了DefaultLitePullConsumerImpl#isSetEqual方法，减少了循环次数和不正确的判断。并添加了isSetEqual方法的单元测试
2. 订正了一个单词拼写错误

How Did You Test This Change?
添加了单元测试验证isSetEqual方法是否符合预期

### CODE REVIEW

File: client/src/test/java/org/apache/rocketmq/client/impl/consumer/DefaultLitePullConsumerImplTest.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * @author Willhow
+ * @since 2024/5/3
+ */
```
Comment (RongtongJin) [inline client/src/test/java/org/apache/rocketmq/client/impl/consumer/DefaultLitePullConsumerImplTest.java:17]: Author information needs to be removed

**Comment by RongtongJin** [comment]
Error:  /home/runner/work/rocketmq/rocketmq/client/src/test/java/org/apache/rocketmq/client/impl/consumer/DefaultLitePullConsumerImplTest.java:1: Line does not match expected header line of '/\*'. [RegexpHeader]
Error:  /home/runner/work/rocketmq/rocketmq/client/src/test/java/org/apache/rocketmq/client/impl/consumer/DefaultLitePullConsumerImplTest.java:15: Recommended remove @author tag in javadoc! [RegexpSingleline]
Error:  /home/runner/work/rocketmq/rocketmq/client/src/test/java/org/apache/rocketmq/client/impl/consumer/DefaultLitePullConsumerImplTest.java:60:91: '{' is not preceded with whitespace. [WhitespaceAround]
Error:  /home/runner/work/rocketmq/rocketmq/client/src/test/java/org/apache/rocketmq/client/impl/consumer/DefaultLitePullConsumerImplTest.java:68:91: '{' is not preceded with whitespace. [WhitespaceAround]
Error:  /home/runner/work/rocketmq/rocketmq/client/src/test/java/org/apache/rocketmq/client/impl/consumer/DefaultLitePullConsumerImplTest.java:76:91: '{' is not preceded with whitespace. [WhitespaceAround]

File: client/src/test/java/org/apache/rocketmq/client/impl/consumer/DefaultLitePullConsumerImplTest.java (hunk 1 above)
Comment (Willhow-Gao) [reply client/src/test/java/org/apache/rocketmq/client/impl/consumer/DefaultLitePullConsumerImplTest.java]: already removed.Idea generated it automatically, so I didn’t notice it.

Comment (Willhow-Gao) [reply client/src/test/java/org/apache/rocketmq/client/impl/consumer/DefaultLitePullConsumerImplTest.java]: > 作者信息需要删除

already removed. Also revised code style and added license comments

**Review by humkum — APPROVED** [review]

**Comment by lizhimins** [comment]
合并一下 develop 重新提交一下，develop 有一些单测的修复

**Comment by Willhow-Gao** [comment]
> 合并一下 develop 重新提交一下，develop 有一些单测的修复

merge develop了

**Review by cserwen — APPROVED** [review]

**Review by RongtongJin — APPROVED** [review]

**Pull request merged**